File: src/Client/src/app/core/signalr/realtime-groups.ts
export const realtimeGroups = {
  categoriesList: 'categories-list',
  categoryImportBatchesList: 'category-import-batches-list',
  itemsList: 'items-list',
  itemImportBatchesList: 'item-import-batches-list',
  goodsReceiptImportsList: 'goods-receipts-import-list',
  schoolClass: (classId: string): `school-class:${string}` => `school-class:${classId}`
} as const;

export type RealtimeGroup =
  | Extract<typeof realtimeGroups[keyof typeof realtimeGroups], string>
  | ReturnType<typeof realtimeGroups.schoolClass>;

File: src/Application/Common/Realtime/RealtimeGroups.cs
namespace skestock.Application.Common.Realtime;

public static class RealtimeGroups
{
    public const string CategoriesList = "categories-list";
    public const string CategoryImportBatchesList = "category-import-batches-list";
    public const string GoodsReceiptImportsList = "goods-receipts-import-list";
    public const string ItemsList = "items-list";
    public const string ItemImportBatchesList = "item-import-batches-list";

    public static string User(string userId) => $"user:{userId}";
    public static string SchoolClass(Guid classId) => $"school-class:{classId}";
}
